component/componenttest/lifecycle_verifier.go

```diff
@@ -0,0 +1,59 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package componenttest
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configmodels"
+)
+
+// GetExtensionConfigFn is used customize the configuration passed to the verification.
+// This is used to change ports or provide values required but not provided by the
+// default configuration.
+type GetExtensionConfigFn func() configmodels.Extension
+
+// VerifyExtensionLifecycle is used to test if an extension type can handle the typical
+// lifecycle of a component. The getConfigFn parameter only need to be specified if
+// the test can't be done with the default configuration for the component.
+func VerifyExtensionLifecycle(t *testing.T, factory component.ExtensionFactory, getConfigFn GetExtensionConfigFn) {
```

Comment:
Can this be a private method in the default configs test? Trying to limit the exposed API, so I am hesitant on exposing these new helpers now that we try to stabilize the API.

Reply:
Sure, I will move it to `defaults_test.go` directly, but, we should consider splitting the test files into `default_extensions_test.go`, `default_exporters_test.go`, and so on; since there should be similar tests for all components.

Reply:
done https://github.com/open-telemetry/opentelemetry-collector/pull/2679/commits/b9a068c15a3d5edee50973dd2596d429fc94f5ba